convert2rhel/unit_tests/subscription_test.py

```diff
@@ -1519,6 +1519,30 @@ def test_update_rhsm_custom_facts_no_rhsm(global_tool_opts, caplog, monkeypatch)
     assert "Skipping updating RHSM custom facts." in caplog.records[-1].message
 
 
+@pytest.mark.parametrize(
+    ("telemetry_disabled", "rhsm_called", "message"),
+    [
+        (True, 0, "Skipping updating RHSM custom facts. Telemetry disabled."),
+        (False, 1, "RHSM custom facts uploaded successfully."),
+    ],
+)
+def test_update_rhsm_custom_facts_disable_telemetry(telemetry_disabled, rhsm_called, message, monkeypatch, caplog):
+    if telemetry_disabled:
+        monkeypatch.setenv("CONVERT2RHEL_DISABLE_TELEMETRY", "1")
+
+    run_subprocess = mock.Mock(
+        side_effect=unit_tests.run_subprocess_side_effect(
+            (["subscription-manager", "facts", "--update"], ("output", 0)),
+        ),
+    )
+    monkeypatch.setattr(utils, "run_subprocess", run_subprocess)
+
+    subscription.update_rhsm_custom_facts()
+
+    assert run_subprocess.call_count == rhsm_called
+    assert message in caplog.records[-1].message
```

Comment:
# WARNING !!
This suggestion follows the code change that I proposed for skipping the RHSM upload. 

```suggestion
def test_update_rhsm_custom_facts_disable_telemetry(monkeypatch, caplog):
    message = "Telemetry disabled, skipping RHSM facts upload."
    monkeypatch.setenv("CONVERT2RHEL_DISABLE_TELEMETRY", "1")
    subscription.update_rhsm_custom_facts()
    assert message in caplog.records[-1].message
```

Pretty more straightforward and only validating that we are skipping the RHSM facts update when the env var is set. As I mentioned, I wouldn't verify other cases, as they are pretty much covered in the functions above this one.

Reply:
Makes sense, simpler test and the condition is easier to understand.